### ISSUE TITLE
INVALID_SENDER - When app is resumed and OneSignal init is called more than once

### ISSUE THREAD

**Issue by jkasten2** [OneSignal/OneSignal-Android-SDK#165]
The following error can be thrown in the logcat if the app is resumed and OneSignal init is called more than once in the app.

```Java
02-01 16:29:54.523 6666-14218/com.onesignal.example E/OneSignal: Error Getting Google Registration ID
                                                                    java.io.IOException: INVALID_SENDER
                                                                        at com.google.android.gms.gcm.GoogleCloudMessaging.zza(Unknown Source)
                                                                        at com.google.android.gms.gcm.GoogleCloudMessaging.register(Unknown Source)
                                                                        at com.onesignal.PushRegistratorGPS$2.run(PushRegistratorGPS.java:151)
                                                                        at java.lang.Thread.run(Thread.java:818)
```

This bug can cause sessions counts to not be counted.

The error is more common in wrapping SDKs (especially Cordova) than it would be in native apps due to OneSignal init being called more than once.

Also if your Google Project number is different in your app than the one on the OneSignal dashboard Android config setting then this could cause `MismatchSenderId` errors from Google when sending a notification. Note in this case you won't see the logcat entry above.

This may show a `INVALID_PARAMETERS` instead of `INVALID_SENDER` in the logcat.

**Comment by yuriploc** [comment]
I'm facing this problem. The OneSignal dashboard shows a `(Google Play Services Library Error)` in the Subscribed column.

Is there any workaround for it?
 
Is this bug possibly related to #146?

**Comment by jkasten2** [comment]
@yuriploc I don't believe this specific #165 issue would cause the `Google Play Services Library Error` error you are seeing. Can you make sure you have entered  a Google Project number under App Settings on the OneSignal dashboard under the Android configuration?

I would also check the logcat for other errors and contact OneSignal support while signed in if you this doesn't point to an issue.

The `Google Play Services Library Error` error is isn't related to issue #146.

**Comment by yuriploc** [comment]
@jkasten2 my bad, the FCM/GCM number was missing on the OneSignal dashboard.
No more problems, the `INVALID_SENDER` IOException doesn't occur anymore and all the users are subscribed.

**Comment by jkasten2** [comment]
Fixed in commit 202351221352b3e8fc5a8b6e3289916a0d601220
